sdk/core/Azure.Core/tests/TestFramework/RecordedTestBase.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using Azure.Core.Pipeline;
+using NUnit.Framework;
+
+namespace Azure.Core.Testing
+{
+    [Category("Recorded")]
+    public abstract class RecordedTestBase: ClientTestBase
+    {
+        private const string ModeEnvironmentVariableName = "AZURE_TEST_MODE";
+
+        protected RecordedTestSanitizer Sanitizer { get; set; }
+
+        protected RecordMatcher Matcher { get; set; }
+
+        protected TestRecording Recording { get; private set; }
+
+        protected RecordedTestBase(bool isAsync) : base(isAsync)
+        {
+            Sanitizer = new RecordedTestSanitizer();
+            Matcher = new RecordMatcher(Sanitizer);
+        }
+
+        protected virtual RecordedTestMode Mode
+        {
+            get
+            {
+                string modeString = Environment.GetEnvironmentVariable(ModeEnvironmentVariableName);
+
+                RecordedTestMode mode = RecordedTestMode.None;
```

Comment:
>RecordedTestMode mode = RecordedTestMode.None; [](start = 16, length = 46)

Perhaps the default mode should be Playback rather than live?  It would make it easier to run all tests, and easier for contributors who don't want to spend money to run tests (or have a subscription capable of running all tests).  However, it does reduce the default coverage and the chances that a dev would opt in to running tests live is low.  Thoughts?

Reply:
I don't have a strong opinion, I left the default the way it was for AppConfiguration. My only argument for having Live be the default is that it's the test with the most coverage.

@tg-msft @jsquire what do you think?

Reply:
Personally, I'd say the expectation from those unfamiliar would be things running against the Live service.   This is how the majority of samples and track one clients are today;  I'd stick with the current default of `None` and make this an opt-in behavior.

Reply:
One thing I'll add is that new users aren't going to jump through the hoops to setup the proper test accounts, etc.  Configuration might not require a lot of setup, but Storage requires multiple accounts with various settings precisely specified to pass.  I see Playback as the default providing a friendlier getting started experience when someone just wants to submit a one line fix.  I'm fine with either Playback or None though.

Reply:
I just added this same question to the main PR comments before seeing this here. I personally would make playback the default mode as well as not require the "Live" category being passed. We want people to be able to clone and build and run tests locally without reading any documents. They cannot run live tests correctly without knowing what other things to configure so I think at that point we can tell them what to configure both for the Azure connections as well as the playback mode. We could even potentially change the mode default based on the presence of some Azure connection strings being set.

Reply:
Done, playback is the default now.